src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/dto/nomis/Mapping.kt

```diff
@@ -0,0 +1,220 @@
+package uk.gov.justice.digital.hmpps.incidentreporting.dto.nomis
+
+import jakarta.validation.ValidationException
+import uk.gov.justice.digital.hmpps.incidentreporting.constants.CorrectionReason
+import uk.gov.justice.digital.hmpps.incidentreporting.constants.InformationSource
+import uk.gov.justice.digital.hmpps.incidentreporting.constants.PrisonerOutcome
+import uk.gov.justice.digital.hmpps.incidentreporting.constants.PrisonerRole
+import uk.gov.justice.digital.hmpps.incidentreporting.constants.StaffRole
+import uk.gov.justice.digital.hmpps.incidentreporting.constants.Status
+import uk.gov.justice.digital.hmpps.incidentreporting.constants.Type
+import uk.gov.justice.digital.hmpps.incidentreporting.jpa.Event
+import uk.gov.justice.digital.hmpps.incidentreporting.jpa.Report
+import java.time.Clock
+import java.time.LocalDateTime
+
+fun NomisReport.toNewEntity(clock: Clock): Report {
+  val report = Report(
+    incidentNumber = "$incidentId",
+    type = mapIncidentType(type),
+    incidentDateAndTime = incidentDateTime,
+    prisonId = prison.code,
+    title = title ?: "NO DETAILS GIVEN",
+    description = description ?: "NO DETAILS GIVEN",
+    reportedBy = reportingStaff.username,
+    reportedDate = reportedDateTime,
+    status = mapIncidentStatus(status.code),
```

Comment:
As mentioned offline, I think it would be _**really nice**_  if the mapping logic were attached to the enums, e.g.

```Kotlin
type = Status.fromCode(status.code)
// type = Type.fromCode(type)
// etc...
```

Reply:
so you want all these string to enum conversions in the same files as the enum classes themselves?